nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -176,10 +184,33 @@ in
 
     systemd.user.services.dbus.wantedBy = [ "default.target" ];
 
-    programs.dconf.profiles.gdm = pkgs.writeText "dconf-gdm-profile" ''
-      system-db:local
-      ${gdm}/share/dconf/profile/gdm
-    '';
+    programs.dconf.profiles.gdm =
+    let
+      customDconf = pkgs.writeTextFile {
+        name = "gdm-dconf";
+        destination = "/dconf/gdm-custom";
+        text = ''
+          ${optionalString (!cfg.gdm.autoSuspend) ''
+            [org/gnome/settings-daemon/plugins/power]
+            sleep-inactive-ac-timeout=0
+            sleep-inactive-battery-timeout=0
```

Comment:
I think we should also set `sleep-inactive-{ac,battery}-type` to `nothing`.

Reply:
Done.

Reply:
pushed https://github.com/NixOS/nixpkgs/pull/63790/commits/1523f0a7bfdbd64ce8ecec4c53e22e016ac12b1b. I meant also.